client/src/modules/user/UserProfile/EmailNotifications.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { PureComponent } from 'react';
+import { injectIntl, FormattedMessage } from 'react-intl';
+
+import AlertBox from 'common/components/AlertBox';
+import SwitchButton from 'common/components/SwitchButton';
+import { IntlPropType } from 'common/constants/propTypes';
+import { saveEmailNotificationSettings } from '../model/actions';
+import { MessageType } from 'common/constants/enums';
+import './EmailNotification.scss';
+
+class EmailNotifications extends PureComponent {
+  state = {
+    error: false,
+    never: false,
+    weekly: true
+  };
+
+  setWeekly = () =>
+    this.setState(
+      { weekly: true, never: false, error: false },
+      this.updateNotificationSettings
+    );
+
+  setNever = () =>
+    this.setState(
+      { never: true, weekly: false, error: false },
+      this.updateNotificationSettings
+    );
+
+  updateNotificationSettings = () => {
+    const { weekly, never } = this.state;
+    const settings = {
+      never,
+      weekly
+    };
+
+    try {
+      saveEmailNotificationSettings(settings);
+    } catch {
+      this.setState({ error: true });
+    }
+  };
+
+  render() {
+    const {
+      intl: { formatMessage }
+    } = this.props;
+    const { error, never, weekly } = this.state;
+
+    return (
+      <section className="email-notifications">
+        {error && (
+          <AlertBox type={MessageType.ERROR}>
```

Comment:
Until now we displayed such errors in our notifications (excluding forms). I don't say that your solution is wrong, maybe it's even better but now it's patchy also